### PR TITLE
Events with favorited button

### PR DESCRIPTION
**Pull request by kirkeaston**

### CODE REVIEW

**Review by calebbratton — APPROVED** [review]

File: client/src/components/Events/Events.jsx [hunk 1]
```diff
@@ -1,8 +1,30 @@
-import React, { useContext, useState } from 'react';
+/* eslint-disable import/no-named-as-default */
+/* eslint-disable import/no-named-as-default-member */
+/* eslint-disable no-trailing-spaces */
+/* eslint-disable no-console */
+/* eslint-disable no-shadow */
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
```
Comment (Pix3lAssassin) [inline client/src/components/Events/Events.jsx]: Seems excessive and should be only used by line unless we really should ignore it

File: client/src/components/Flights/Flights.jsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-unused-vars */
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/Flights.jsx]: Should remove this

File: client/src/components/Tabs.jsx [hunk 3]
```diff
@@ -1,3 +1,7 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/require-default-props */
```
Comment (Pix3lAssassin) [inline client/src/components/Tabs.jsx]: Should remove most of these or make them single liners

File: server/routes/flights.js [hunk 4]
```diff
@@ -166,3 +153,63 @@ module.exports = router;
...
+//       },
+//     }),
+//   )).then((response) => {
+//     res.status(200).send(response.result);
+//   })
+//     .catch((response) => {
+//       console.log('error posting booking', response);
+//       res.status(500).send(response);
+//     });
+// });
```
Comment (Pix3lAssassin) [inline server/routes/flights.js:215]: Should be removed as soon as current /booking is tested

File: client/src/components/Events/Entry.jsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/alt-text */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable react/prop-types */
+/* eslint-disable no-unused-vars */
+/* eslint-disable eol-last */
```
Comment (Pix3lAssassin) [inline client/src/components/Events/Entry.jsx:7]: Excessive disabling

File: client/src/components/Events/Entry.jsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+import React, { useContext, useState, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import ButtonBase from '@material-ui/core/ButtonBase';
+import FavoriteIcon from '@material-ui/icons/Favorite';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+import { Fab } from '@material-ui/core/';
```
Comment (Pix3lAssassin) [inline client/src/components/Events/Entry.jsx]: Could probably use object destructuring for the core items

File: client/src/components/Events/Entry.jsx [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+  img: {
+    margin: 'auto',
+    display: 'block',
+    maxWidth: '100%',
+    maxHeight: '100%',
+  },
+  input: {
+    display: 'none',
+  },
+}));
```
Comment (Pix3lAssassin) [inline client/src/components/Events/Entry.jsx:41]: P3 fix: extract this styling to a new file

File: client/src/components/Events/EventForm.jsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable no-unused-vars */
```
Comment (Pix3lAssassin) [inline client/src/components/Events/EventForm.jsx]: Should probably remove these

File: client/src/components/Events/EventForm.jsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+import React, { useContext, useState, useEffect } from 'react';
```
Comment (Pix3lAssassin) [inline client/src/components/Events/EventForm.jsx:2]: useState and useEffect are unused

File: client/src/components/Events/Events.jsx [hunk 10]
```diff
@@ -19,13 +41,70 @@ const useStyles = makeStyles((theme) => ({
...
+    axios.get('/api/events', {
+      params: {
+        latitude,
+        longitude,
+        category,
+      },
+    })
+      .then((result) => {
+        updateEventData(result.data.data);
+        console.log(result.data.data);
```
Comment (Pix3lAssassin) [inline client/src/components/Events/Events.jsx]: should probably remove this console.log

File: client/src/components/Events/Entry.jsx [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+                        {' '}
+                        {eventTags[1]}
+                      </span>
+                    ) : null}
+                  {eventTags[2]
+                    ? (
+                      <span>
+                        ,
+                        {' '}
+                        {eventTags[2]}
```
Comment (Pix3lAssassin) [inline client/src/components/Events/Entry.jsx:105]: should use string templating here

File: client/src/components/Events/Entry.jsx [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+    const firstFour = array.slice(0, 4);
+    return firstFour;
+  };
+  const eventTags = getOnlyFirstFour(tags);
+  let source;
+  const eventIcons = (cat) => {
+    cat === 'RESTAURANT' ? source = 'https://www.creativefabrica.com/wp-content/uploads/2018/12/Restaurant-icon-vector-by-Hoeda80-580x386.jpg' : null;
+    cat === 'NIGHTLIFE' ? source = 'https://image.shutterstock.com/image-illustration/dancing-couple-icon-element-party-260nw-1159039519.jpg' : null;
+    cat === 'SIGHTS' ? source = 'https://image.shutterstock.com/image-vector/eye-icon-vector-260nw-1329652607.jpg' : null;
+    cat === 'SHOPPING' ? source = 'https://image.shutterstock.com/image-vector/shopping-bag-vector-icon-260nw-661740685.jpg' : null;
```
Comment (Pix3lAssassin) [inline client/src/components/Events/Entry.jsx:59]: could probably make this into an object with keys to improve readablility

File: client/src/components/Events/Events.jsx (hunk 1 above)
Comment (kirkeaston) [reply client/src/components/Events/Events.jsx]: Changed them to by line

File: client/src/components/Flights/Flights.jsx (hunk 2 above)
Comment (kirkeaston) [reply client/src/components/Flights/Flights.jsx]: removed!

File: client/src/components/Tabs.jsx (hunk 3 above)
Comment (kirkeaston) [reply client/src/components/Tabs.jsx]: removed!

File: client/src/components/Events/Entry.jsx (hunk 6 above)
Comment (kirkeaston) [reply client/src/components/Events/Entry.jsx]: destructured!

**Review by Pix3lAssassin — APPROVED** [review]

**Pull request merged**